wgengine/router/osrouter/router_linux.go

```diff
@@ -540,6 +545,8 @@ func (r *linuxRouter) updateStatefulFilteringWithDockerWarning(cfg *router.Confi
 
 // UpdateMagicsockPort implements the Router interface.
 func (r *linuxRouter) UpdateMagicsockPort(port uint16, network string) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
 	if r.nfr == nil {
 		if err := r.setupNetfilter(r.netfilterKind); err != nil {
```

Comment:
I think we cannot do this here, can we? `setupNetfilter` also acquires this lock I believe.